Add initial docs for `bfloat16` and `on_disk_rescore`  for dense vectors (#3847)
elastic/docs-content: solutions/search/vector/knn.md

}
```

### BFloat16 vector encoding [knn-search-bfloat16]
```
{applies_to}
stack: ga 9.3
```

> Instead of storing raw vectors as 4-byte values, you can use `element_type: bfloat16` to store each dimension as a 2-byte value. This can be useful if your indexed vectors are at bfloat16 precision already, or if you wish to reduce the disk space required to store vector data. Elasticsearch will automatically truncate 4-byte float values to 2-byte bfloat16 values when indexing vectors.

Reviewer: ```suggestion
Instead of storing raw vectors as 4-byte values, you can use `element_type: bfloat16` to store each dimension as a 2-byte value. This can be useful if your indexed vectors are at bfloat16 precision already, or if you want to reduce the disk space required to store vector data. When this element type is used, {{es}} automatically truncates 4-byte float values to 2-byte bfloat16 values when indexing vectors.
```

* Return the top 10 (`k`) rescored candidates.
* Merge the rescored candidates from all shards, and return the top 10 (`k`) results.

#### The `on_disk_rescore` option
```{applies_to}
stack: preview 9.3
serverless: unavailable
```

By default, Elasticsearch will read raw vector data into memory to perform rescoring. This may have an effect on performance if the vector data is too large to all fit in off-heap memory at once. By specifying the `on_disk_rescore: true` index setting, Elasticsearch will read vector data from disk directly during rescoring.

> Note that this setting will only apply to newly indexed vectors; to apply the option to all vectors in the index, the vectors must be re-indexed or force-merged after changing the setting.

Reviewer: ```suggestion
This setting only applies to newly indexed vectors. To apply the option to all vectors in the index, the vectors must be re-indexed or force-merged after changing the setting.
```
